flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/JsonRowDeserializationSchemaTest.java

```diff
@@ -134,7 +135,8 @@ public void testTypeInfoDeserialization() throws Exception {
         row.setField(9, map);
         row.setField(10, nestedMap);
 
-        assertThat(serializedJson, whenDeserializedWith(deserializationSchema).equalsTo(row));
+        assertThat(serializedJson)
+                .satisfies(matching(whenDeserializedWith(deserializationSchema).equalsTo(row)));
```

Comment:
I believe the main purpose for the existence of `whenDeserializedWith` was to make the test readable. The current construct has lost that property and using this matcher does not seem to bring much benefit. We should either convert it into a static utility method or consider writing a custom assertion
https://joel-costigliola.github.io/assertj/assertj-core-custom-assertions.html